mutatex/utils.py

```diff
@@ -421,43 +422,59 @@ def get_foldx_sequence(pdb, multimers=True):
+            positions[chain_name] = ''
             for residue in chain:
                 try:
                     res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
+                    pos_code = str(residue.get_id()[1])
                 except:
                     log.warning("Residue %s in file %s couldn't be recognized; it will be skipped" %(residue, pdb))
                     continue
                 if not multimers:
                     residue_list.append(tuple(["%s%s%d" % (res_code, chain.get_id(), residue.get_id()[1])]))
                 else:
                     sequences[chain_name] += res_code
+                    positions[chain_name] += pos_code
 
     if multimers:
-        collated_chains = []
+        collated_chains_pos = []
+        collated_chains_seq = []
         seq_ids, seqs = list(zip(*list(iteritems(sequences))))
+        pos_ids, pos = list(zip(*list(iteritems(positions))))
         seq_ids = np.array(seq_ids)
+        pos_ids = np.array(pos_ids)
+
         unique_seqs, unique_idxs = np.unique(seqs, return_inverse=True)
+        unique_pos, unique_idxp = np.unique(pos, return_inverse=True)
 
+        for i in np.unique(unique_idxp):
+            collated_chains_pos.append(pos_ids[unique_idxp == i])
         for i in np.unique(unique_idxs):
-            collated_chains.append(seq_ids[unique_idxs == i])
+            collated_chains_seq.append(seq_ids[unique_idxs == i])
+
+        if len(collated_chains_seq) > len(collated_chains_pos) or len(collated_chains_seq) == len(collated_chains_pos):
+            for cg in collated_chains_seq:
+                for model in structure:
+                    for residue in model[cg[0]]:
+                        resid = residue.get_id()[1]
+                        try:
+                            res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
+                        except:
+                            log.warning("Residue %s in file %s couldn't be recognized; it will be skipped" %(residue, pdb))
+                            continue
+                        this_res = tuple(sorted([ "%s%s%d" % (res_code, c, resid) for c in cg ], key=lambda x: x[1]))
+                        residue_list.append(this_res)
+            return tuple(residue_list)
 
-        for cg in collated_chains:
-            for model in structure:
-                for residue in model[cg[0]]:
-                    resid = residue.get_id()[1]
-                    try:
-                        res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
-                    except:
-                        log.warning("Residue %s in file %s couldn't be recognized; it will be skipped" %(residue, pdb))
-                        continue
-                    this_res = tuple(sorted([ "%s%s%d" % (res_code, c, resid) for c in cg ], key=lambda x: x[1]))
-                    residue_list.append(this_res)
-
-    return tuple(residue_list)
+        else:
```

Comment:
I see what you're trying to do here, but I would prefer if our function would return always the same data structure, otherwise it becomes complicated to document/predict its behaviour, also this is not a very pythonic way to communicate that something went wrong and/or is unexpected, that would be by using exceptions